doc/scripts/coccinelle.rst

```diff
@@ -0,0 +1,487 @@
+.. Copyright 2010 Nicolas Palix <npalix@diku.dk>
```

Comment:
Check out the Zephyr doc writing conventions at http://docs.zephyrproject.org/contribute/doc-guidelines.html.  For example, documents start with a label that allows cross-document linking:

    .. _coccinelle:

And our heading hierarchy is #, *, =, -

That aside, I'm not sure this is the right level of documentation for the Zephyr project.  It makes assumptions that the reader knows about Coccinelle and doesn't explain why a Zephyr developer should care or want to use this (external) tool?  What does it tell them about the Zephyr code it's scanning?

The examples and explanations are all specific to the Linux kernel and not the Zephyr project.  (If this is just a copy of Coccinelle documentation, then it would be better to have some Zephyr-specific information here, and point to the details over in your documentation.)  Examples here should be Zephyr-specific and help explain the value of this tool to the Zephyr developer.

Reply:
The ``doc/scripts`` folder is where we have scripts used to prepare documentation. Your documentation would be better placed in the ``doc/application`` folder and linked into the "Developer Guides" TOC (the index.rst file that's in the doc/application folder.